### DISCUSSION TITLE
GitHub Copilot in VS2022 hung while running in not very good network conditions

### DISCUSSION THREAD

**Discussion by BaZzz01010101** [community/community#116906] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Recently, my ISP has started experiencing issues with packet loss on some connections. Specifically, copilot-proxy.githubusercontent.com has up to 15% packet loss. At the same time, I noticed that the Copilot extension in my VS2022 became almost unusable. It works for a very short time after Visual Studio starts and then stops working. Sometimes, Copilot recovers if I don't use VS for a long time, but usually only restarting VS helps. However, this only resolves the issue for a few minutes.

The Copilot Log does not contain any errors or warnings, only lines like this:
```
[DEBUG] [getCompletions] [2024-04-05T10:14:14.611Z] Requesting completion at position 708:0, between "        .secondaryFire = true,\r\n" and "\r\n      };\r\n".
[INFO] [default] [2024-04-05T10:14:14.721Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[DEBUG] [getCompletions] [2024-04-05T10:14:24.036Z] Requesting completion at position 708:0, between "        .secondaryFire = true,\r\n" and "\r\n      };\r\n".
[INFO] [default] [2024-04-05T10:14:24.132Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[DEBUG] [getCompletions] [2024-04-05T10:14:45.098Z] Requesting completion at position 708:0, between "        .secondaryFire = true,\r\n" and "\r\n      };\r\n".
[INFO] [default] [2024-04-05T10:14:45.188Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
```

I assume that the extension has a buffer for API requests. The extension is constantly sending requests for code suggestions. If this buffer doesn't have enough size and the timeout is too high, it could overflow due to lost responses. This could explain the problem. 

Is there any chance to get some insights from the development team?

**Comment by ZeEitan** [comment]
Hi, did you manage to fix it? I have the same problem

**Reply by ayim** [reply]
Hey folks, Product Manager from the development team here.
@ZeEitan could you...
1) attach some logs like the user did above?
2) file a bug in Visual Studio using the "Report a Problem" feature

![image](https://github.com/community/community/assets/10951869/1bc11412-a3a7-4f11-ad1b-9fa372ef819c)

**Reply by ZeEitan** [reply]
Yes, no problem, thank you very much

**Reply by ZeEitan** [reply]
![image](https://github.com/community/community/assets/113852982/6dd78e6a-755e-4e90-9090-65a274ec2645)
I have a problem with the report 😂

**Reply by ayim** [reply]
^Could you try this maybe 5-10X? I find that it sometimes doesn't catch on the first try depending on the browser that I'm using.

**Reply by ZeEitan** [reply]
I sent a report

**Reply by BaZzz01010101** [reply]
> Hi, did you manage to fix it? I have the same problem

No I didn't. The Copilot is not usable 95% of the time. I'm thinking about to change my ISP but Copilot should definitely work in such conditions.

**Comment by ayim** [comment]
@BaZzz01010101 interesting that this is working in the first few moments that you launch Visual Studio.
We need more diagnostics details to investigate.. could you

1) file a bug in Visual Studio using the "Report a Problem" feature
![image](https://github.com/community/community/assets/10951869/a34c04f8-e309-4025-9242-292b1c651976)
2) Attach logs to the bug

Then just in case we need to contact you for live demo 
3) Send me an email at aayim@microsoft.com?

**Reply by BaZzz01010101** [reply]
Done
https://developercommunity.visualstudio.com/t/GitHub-Copilot-stop-working-when-connect/10643366
